examples/blockchain/network.rs

```diff
@@ -0,0 +1,377 @@
+use crate::chain::Block;
+use aleph_bft::NodeIndex;
+use codec::{Decode, Encode};
+use futures::{
+    channel::{
+        mpsc::{self, UnboundedReceiver, UnboundedSender},
+        oneshot,
+    },
+    prelude::*,
+    Future, FutureExt, StreamExt,
+};
+
+use log::{debug, info};
+
+use std::{collections::HashMap, convert::TryFrom as _, error::Error, io, iter, time::Duration};
+
+use libp2p::{
+    development_transport, identity,
+    mdns::{Mdns, MdnsConfig, MdnsEvent},
+    request_response::{
+        ProtocolSupport, RequestResponse, RequestResponseCodec, RequestResponseConfig,
+        RequestResponseEvent, RequestResponseMessage,
+    },
+    swarm::{NetworkBehaviourEventProcess, SwarmBuilder},
+    NetworkBehaviour, PeerId, Swarm,
+};
+
+use crate::{
+    chain::Data,
+    crypto::{Hasher256, PartialMultisignature, Signature},
+};
+
+#[derive(Clone)]
+pub(crate) struct Spawner;
+
+impl aleph_bft::SpawnHandle for Spawner {
+    fn spawn(&self, _: &str, task: impl Future<Output = ()> + Send + 'static) {
+        tokio::spawn(task);
+    }
+}
+
+const ALEPH_PROTOCOL_NAME: &str = "/aleph/chain/1";
+
+type NetworkData = aleph_bft::NetworkData<Hasher256, Data, Signature, PartialMultisignature>;
+
+#[derive(Clone, Encode, Decode)]
+enum Message {
+    Auth(NodeIndex),
+    Consensus(Vec<u8>),
```

Comment:
I'm pretty sure the `Vec<u8>` here could just be `NetworkData` with no additional effort. This saves one `encode` call every time this is used, but more importantly just seems cleaner – consensus messages are not arbitrary byte sequences after all.

Unless I missed some reason you actually need this to be a sequence of bytes?

Reply:
You are absolutely right. I fixed it. 

However, after fixing it, clippy is unhappy because the Consensus() enum variant is huge -- it says 384 bytes (why lol?) and there is this lint https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant -- so I put an ignore for this here. The ignore is fine, as anyway 99% of the messages are consensus messages.

Reply:
Ah, interesting. That is a reason to have it be `Vec<u8>` I did not know about although, as you say, it is not a strong reason in this case.